t/16-utils-runcmd.t

```diff
@@ -143,16 +129,47 @@ subtest 'git commands with mocked run_cmd_with_log_return_error' => sub {
             [qw(git -C /repo/path rm bar.png bar.json)],
             [
                 qw(git -C /repo/path commit -q -m),
-                'some test',
+                'add rm test',
                 '--author=openQA system user <noemail@open.qa>',
                 qw(foo.png foo.json bar.png bar.json)
             ],
         ],
         'changes staged and committed',
     ) or diag explain \@executed_commands;
+
+    $git->config->{do_push} = 'yes';
+
+    local $mock_return_value{status} = 1;
+    local $mock_return_value{stderr} = 'mocked push error';
+    local $mock_return_value{stdout} = '';
+    
+    $utils_mock->redefine(
+    run_cmd_with_log_return_error => sub {
+        my ($cmd) = @_;
+        push(@executed_commands, $cmd);
+        if ($cmd->[3] eq 'push') {
+          $mock_return_value{status} = 0;
+        }
+        return \%mock_return_value;
+    });
+    is(
+        $git->commit(
+            {
+                message => 'failed push test',
+            }
+        ),
+        'Unable to push Git commit: mocked push error',
+        'an error occurred during push'
+    );
```

Comment:
That's using a lot of space.
```suggestion
    is(
        $git->commit({message => 'failed push test'}),
        'Unable to push Git commit: mocked push error',
        'an error occurred during push'
    );
```

Reply:
```suggestion
    like $git->commit( {message => 'failed push test'}), qr/Unable to push Git commit/, 'error handled during push';
```

because shorter is more readable

Reply:
shorter is not always more readable. that's highly subjective.
but in the end it depends on perltidy.